scripts/spec_xml/paths.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from enum import Enum
+
+# Define a branch enum for different versions or branches
+
+
+class Branch(Enum):
+    MASTER = "master"
+    V1_3 = "v1_3"
+    V1_4 = "v1_4"
+    IN_PROGRESS = "in_progress"
+
+
+def get_chip_root():
```

Comment:
HIT: How about returning `Path` object here (and in the rest of these path helpers)?

https://docs.python.org/3/library/pathlib.html

Then you will be able to semantically operate on the path object instead of using `os.path.join` or other `os.path...`. Since we are on Python >= 3.9 there are a lot of goodies in the pathlib library.

Reply:
Let's update that one as a followup due to the sizes of the PR and it being open for a very long time (i.e. harder to iterate on it)